packages/flutter/lib/src/services/text_input.dart

```diff
@@ -874,7 +874,13 @@ class TextEditingValue {
     final String newText = text.replaceRange(replacementRange.start, replacementRange.end, replacementString);
 
     if (replacementRange.end - replacementRange.start == replacementString.length) {
-      return copyWith(text: newText);
+      return copyWith(
```

Comment:
I think the selection needs to be preserved here:
>   /// This method also adjusts the selection range and the composing range of the
  /// resulting [TextEditingValue], such that they point to the same substrings
  /// as the correspoinding ranges in the original [TextEditingValue].

For instance `FilteringTextInputFormatter` uses this. And if I copy "hello word" into a text field and a `FilteringTextInputFormatter` changes "hello" into "olleh", the caret should still stay at the end of the whole string, instead of moving to the position immediately after "olleh".

Reply:
Good call, I was hopeful that the selection should have gone at the end all along when I saw that tests still passed.  I've updated it to no longer use this Action and manually make the change instead.